From: Chris Lamb <lamby@debian.org>
Date: Fri, 30 Oct 2015 10:53:42 +0000
Subject: Fix FTBFS on kFreeBSD

---
 src/fmacros.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fmacros.h b/src/fmacros.h
index d05230d..3d6b7c9 100644
--- a/src/fmacros.h
+++ b/src/fmacros.h
@@ -21,7 +21,7 @@
 #define _ALL_SOURCE
 #endif
 
-#if defined(__linux__) || defined(__OpenBSD__)
+#if defined(__linux__) || defined(__OpenBSD__) || defined(__GLIBC__)
 #define _XOPEN_SOURCE 700
 /*
  * On NetBSD, _XOPEN_SOURCE undefines _NETBSD_SOURCE and
