From: Chris Lamb <lamby@debian.org>
Date: Wed, 24 Jan 2018 22:06:35 +1100
Subject: Use get_current_dir_name over PATHMAX, etc.

---
 src/rdb.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/rdb.c b/src/rdb.c
index 1538039..2128d13 100644
--- a/src/rdb.c
+++ b/src/rdb.c
@@ -1526,7 +1526,6 @@ werr: /* Write error. */
 }
 
 static int rdbSaveInternal(int req, const char *filename, rdbSaveInfo *rsi, int rdbflags) {
-    char cwd[MAXPATHLEN]; /* Current working dir path for error messages. */
     rio rdb;
     int error = 0;
     int saved_errno;
@@ -1536,7 +1535,7 @@ static int rdbSaveInternal(int req, const char *filename, rdbSaveInfo *rsi, int
     if (!fp) {
         saved_errno = errno;
         char *str_err = strerror(errno);
-        char *cwdp = getcwd(cwd,MAXPATHLEN);
+        char *cwdp = get_current_dir_name();
         serverLog(LL_WARNING,
             "Failed opening the temp RDB file %s (in server root dir %s) "
             "for saving: %s",
@@ -1598,7 +1597,6 @@ int rdbSaveToFile(const char *filename) {
 /* Save the DB on disk. Return C_ERR on error, C_OK on success. */
 int rdbSave(int req, char *filename, rdbSaveInfo *rsi, int rdbflags) {
     char tmpfile[256];
-    char cwd[MAXPATHLEN]; /* Current working dir path for error messages. */
 
     startSaving(rdbflags);
     snprintf(tmpfile,256,"temp-%d.rdb", (int) getpid());
@@ -1612,7 +1610,7 @@ int rdbSave(int req, char *filename, rdbSaveInfo *rsi, int rdbflags) {
      * if the generate DB file is ok. */
     if (rename(tmpfile,filename) == -1) {
         char *str_err = strerror(errno);
-        char *cwdp = getcwd(cwd,MAXPATHLEN);
+        char *cwdp = get_current_dir_name();
         serverLog(LL_WARNING,
             "Error moving temp DB file %s on the final "
             "destination %s (in server root dir %s): %s",
